src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+
+/**
+ * This listener allows you to inspect all exceptions thrown by the application.
+ * This is useful since the HttpKernel catches all exceptions and turn them into
+ * an HTTP response.
+ *
+ * This class should only be used in tests.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ExceptionListener
+{
+    /**
+     * @var list<\Throwable>
+     */
+    private static array $exceptions = [];
+
+    public function __invoke(ExceptionEvent $event)
+    {
+        self::$exceptions[] = $event->getThrowable();
+    }
+
+    /**
+     * @return list<\Throwable>
+     */
+    public static function getExceptions(): array
+    {
+        $exceptions = self::$exceptions;
+        self::$exceptions = [];
```

Comment:
For a getter method I would not expect the method to modify the object's state but to be able to call the method several times with the same result.

Reply:
I'll name this `resetExceptions()`

Reply:
Now it's not really clear that I could use this method retrieve the list of all caught exceptions. What about adapting the schema used for methods in the SPL and use something like `shiftAll()`?